### PR TITLE
Fixing issues in the Jupyter notebook for the planner

### PR DESCRIPTION
**Pull request by kaza**
### Motivation and Context

This change fixes the issue in dotnet Jupyternotebooks, for the planner examples
It useses different classes for the planer, and in addition it fixes few smaller issues regarding configuration and presenting of the resultes

### Description

1)  Changed the  used nuget version
2)  Changed  the old PlannerSkill class to SequentialPlanner
3)  local config files, for azure and open AI, both must have all the properties as otherwise Json deserializer throws an exception, as I used manually changed config files, had issue in configuring both of them, so I added missing json properties
4) in the Planner originalPlan.Variables.ToPlan().PlanString has been removed so I created a new method in Utils and used that one

### CODE REVIEW

**Comment by alexchaomander** [comment]
Thank you so much for creating this! This notebook needed to get updated to reflect the latest planner refactor. One comment from my side, for this particular example, could you show the output of the cells in the Jupyter notebook? Since it's a popular notebook that a lot of people like to look at (even on the Github website), it would be good to see what a "plan" looks like.

Thanks again!

@lemillermicrosoft  @RogerBarreto can you take a look?

**Comment by dluc** [comment]
@kaza thanks for the PR, I'll leave some comments

**Comment by dluc** [comment]
@kaza could you allow us to edit the PR of minor adjustments before merging?

also, could you "Clear All Outputs" ?

image

I'd be glad to do it but we cannot edit the PR

**Comment by kaza** [comment]
I cleared all the outputs hover previously I was asked to inoclude them "could you show the output of the cells in the Jupyter notebook?". 
If you want I can generate them and include them again :)

File: samples/notebooks/dotnet/config/settings.json.azure-example [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "type":     "azure",
   "model":    "... your Azure OpenAI deployment name ...",
   "endpoint": "https:// ... your endpoint ... .openai.azure.com/",
-  "apikey":   "... your Azure OpenAI key ..."
+  "apikey":   "... your Azure OpenAI key ...", 
+  "org":      "" // it's not used for azure, but the parser requires it so do not delete
```
Comment (dluc) [inline samples/notebooks/dotnet/config/settings.json.azure-example:6]: interesting, looks like a bug I've never hit. Thanks!

**Comment by dluc** [comment]
I would remove the custom serialization logic before merging, and leverage .NET JSON serializer + the fact that the Plan data structure is serializable

File: samples/notebooks/dotnet/config/Utils.cs [hunk 2]
```diff
@@ -22,4 +22,39 @@ public static string WordWrap(string text, int maxLineLength)
 
         return result.ToString();
     }
+
```
[the remark this answers is not in the document]
Comment (lemillermicrosoft) [reply samples/notebooks/dotnet/config/Utils.cs:25]: I do like the formatting personally, especially in a notebook experience. Can understand both perspectives.

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 3]
```diff
@@ -67,7 +71,7 @@
    "outputs": [],
    "source": [
     "// Load native skill into the kernel registry, sharing its functions with prompt templates\n",
-    "var planner = kernel.ImportSkill(new PlannerSkill(kernel));"
+    " var planner = new SequentialPlanner(kernel);"
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:74]:
```suggestion
    "var planner = new SequentialPlanner(kernel);"
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 4]
```diff
@@ -25,23 +25,27 @@
...
     "// Configure AI backend used by the kernel\n",
     "var (useAzureOpenAI, model, azureEndpoint, apiKey, orgId) = Settings.LoadFromFile();\n",
+    "\n",
+    "\n",
     "if (useAzureOpenAI)\n",
-    "    kernel.Config.AddAzureTextCompletionService(\"davinci\", model, azureEndpoint, apiKey);\n",
+    "    kernel.Config.AddAzureTextCompletionService( model, azureEndpoint, apiKey);\n",
     "else\n",
-    "    kernel.Config.AddOpenAITextCompletionService(\"davinci\", model, apiKey, orgId);"
+    "    kernel.Config.AddOpenAITextCompletionService( model, apiKey, orgId);\n"
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:48]:
```suggestion
    "    kernel.Config.AddOpenAITextCompletionService( model, apiKey, orgId);"
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 5]
```diff
@@ -25,23 +25,27 @@
...
-    "using Microsoft.SemanticKernel.KernelExtensions;\n",
     "using Microsoft.SemanticKernel.Orchestration;\n",
+    "using Microsoft.SemanticKernel.Planning;\n",
+    "using Microsoft.SemanticKernel.Planning.Sequential;\n",
     "\n",
     "IKernel kernel = KernelBuilder.Create();\n",
     "\n",
     "// Configure AI backend used by the kernel\n",
     "var (useAzureOpenAI, model, azureEndpoint, apiKey, orgId) = Settings.LoadFromFile();\n",
+    "\n",
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:43]:
```suggestion
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 6]
```diff
@@ -116,30 +119,14 @@
...
+   "outputs": [],
    "source": [
     "var ask = \"Tomorrow is Valentine's day. I need to come up with a few date ideas and e-mail them to my significant other.\";\n",
-    "var originalPlan = await kernel.RunAsync(ask, planner[\"CreatePlan\"]);\n",
+    "var originalPlan = await planner.CreatePlanAsync(ask);\n",
     "\n",
     "Console.WriteLine(\"Original plan:\\n\");\n",
-    "Console.WriteLine(originalPlan.Variables.ToPlan().PlanString);"
+    "Console.WriteLine(JsonSerializer.Serialize(originalPlan, new JsonSerializerOptions { WriteIndented = true }));\n",
+    "\n"
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:129]:
```suggestion
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 7]
```diff
@@ -116,30 +119,13 @@
...
-   ],
+   "outputs": [],
    "source": [
     "var ask = \"Tomorrow is Valentine's day. I need to come up with a few date ideas and e-mail them to my significant other.\";\n",
-    "var originalPlan = await kernel.RunAsync(ask, planner[\"CreatePlan\"]);\n",
+    "var originalPlan = await planner.CreatePlanAsync(ask);\n",
     "\n",
     "Console.WriteLine(\"Original plan:\\n\");\n",
-    "Console.WriteLine(originalPlan.Variables.ToPlan().PlanString);"
+    "Console.WriteLine(JsonSerializer.Serialize(originalPlan, new JsonSerializerOptions { WriteIndented = true }));\n",
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:128]:
```suggestion
    "Console.WriteLine(JsonSerializer.Serialize(originalPlan, new JsonSerializerOptions { WriteIndented = true }));",
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 8]
```diff
@@ -206,23 +192,20 @@
    "outputs": [],
    "source": [
     "var ask = @\"Tomorrow is Valentine's day. I need to come up with a few date ideas.\n",
-    " She likes Shakespeare so write using his style. E-mail these ideas to my significant other\";"
+    " She likes Shakespeare so write using his style. E-mail these ideas to my significant other\";\n",
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:195]:
```suggestion
    "She likes Shakespeare so write using his style. E-mail these ideas to my significant other\";\n",
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 9]
```diff
@@ -283,106 +249,9 @@
...
-    "        break;\n",
-    "    }\n",
-    "    \n",
-    "    executionResults = results;\n",
-    "    step++;\n",
-    "    Console.WriteLine(\"\");\n",
-    "}"
+    "var newPlanResult = await kernel.RunAsync(newPlan);\n",
+    "Console.WriteLine(\"New  Plan  results:\\n\");\n",
+    "Console.WriteLine(newPlanResult.Result);\n"
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:254]:
```suggestion
    "var newPlanResult = await kernel.RunAsync(newPlan);\n",
    "Console.WriteLine(\"New Plan results:\\n\");\n",
    "Console.WriteLine(newPlanResult.Result);"
```

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 10]
```diff
@@ -236,38 +219,21 @@
...
-   ],
+   "outputs": [],
    "source": [
-    "Console.WriteLine(\"Updated plan:\\n\");\n",
-    "Console.WriteLine(newPlan.Variables.ToPlan().PlanString);"
+    "var originalPlanResult = await kernel.RunAsync(originalPlan);\n",
+    "\n",
+    "Console.WriteLine(\"Original Plan  results:\\n\");\n",
+    "Console.WriteLine(Utils.WordWrap(originalPlanResult.Result, 80));\n",
+    "\n"
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:228]:
```suggestion
    "var originalPlanResult = await kernel.RunAsync(originalPlan);\n",
    "\n",
    "Console.WriteLine(\"Original Plan results:\\n\");\n",
    "Console.WriteLine(Utils.WordWrap(originalPlanResult.Result, 80));"
```

**Review by dluc — CHANGES_REQUESTED** [review]

File: samples/notebooks/dotnet/05-using-the-planner.ipynb [hunk 11]
```diff
@@ -116,30 +119,13 @@
...
-   ],
+   "outputs": [],
    "source": [
     "var ask = \"Tomorrow is Valentine's day. I need to come up with a few date ideas and e-mail them to my significant other.\";\n",
-    "var originalPlan = await kernel.RunAsync(ask, planner[\"CreatePlan\"]);\n",
+    "var originalPlan = await planner.CreatePlanAsync(ask);\n",
     "\n",
     "Console.WriteLine(\"Original plan:\\n\");\n",
-    "Console.WriteLine(originalPlan.Variables.ToPlan().PlanString);"
+    "Console.WriteLine(JsonSerializer.Serialize(originalPlan, new JsonSerializerOptions { WriteIndented = true }));",
```
Comment (dluc) [inline samples/notebooks/dotnet/05-using-the-planner.ipynb:128]:
```suggestion
    "Console.WriteLine(JsonSerializer.Serialize(originalPlan, new JsonSerializerOptions { WriteIndented = true }));"
```

**Review by dluc — APPROVED** [review]

**Pull request merged**